File: stacked-ios/Stacked/DesignSystem/PaletteColors.swift
import SwiftUI

// Paridade stacked-web/lib/theme/palette-colors.ts
// Ordem: famílias por matiz (claro → médio → saturado → escuro), depois neon.
enum PaletteColors {
  static let projectHex: [String] = [
    // Vermelhos
    "#FDA4AF", "#F87171", "#EF5A5F", "#F43F5E", "#EF4444",
    "#DC2626", "#BE123C", "#B91C1C", "#7F1D1D",
    // Laranjas / âmbar
    "#FCD34D", "#FBBF24", "#FFD166", "#F5A623", "#F59E0B",
    "#FB923C", "#F97316", "#E07B54", "#C58A72", "#EA580C",
    "#B45309", "#92400E", "#7C2D12",
    // Amarelos / lima
    "#D3B36A", "#C7B38A", "#84CC16", "#365314",
    // Verdes
    "#86EFAC", "#4ADE80", "#8FD46B", "#34D399", "#64D8A0",
    "#22C55E", "#10B981", "#84B98E", "#82B392", "#6D9E82",
    "#7FA88E", "#93A882", "#5F9688", "#6B8F7A", "#5B8A70",
    "#789C6B", "#4A7359", "#15803D", "#047857", "#166534",
    // Teals / cianos
    "#63C7D8", "#5FD3DC", "#06B6D4", "#0E7490", "#0F766E", "#134E4A",
    // Azuis
    "#93C5FD", "#60A5FA", "#9CB8D4", "#8DB4CC", "#4D9FEC",
    "#3B82F6", "#7A9BB8", "#6A8FAE", "#6F8FB8", "#5E89A8",
    "#7F99A8", "#4B7294", "#556B8A", "#3E6582", "#2563EB",
    "#1D4ED8", "#1E40AF",
    // Índigos / roxos
    "#E9D5FF", "#DDD6FE", "#C4B5FD", "#A496C8", "#B18CF5",
    "#A78BFA", "#8B5CF6", "#6F79B6", "#7C3AED", "#6D28D9",
    "#D946EF", "#5B21B6", "#312E81",
    // Rosas / magenta
    "#F9A8D4", "#F0ABFC", "#FF85A1", "#FB7185", "#F472B6",
    "#E879F9", "#EC4899", "#C58D97", "#DB2777", "#BE185D",
    "#831843", "#701A75",
    // Cinzas
    "#D1D5DB", "#B8BCC4", "#A3A9B2", "#9CA3AF", "#949BA6",
    "#8A9099", "#727884", "#6E737C", "#565C66", "#4B5563", "#374151",
    // Neon / elétrico
    "#FF073A", "#FF5E00", "#FFFF33", "#CCFF00", "#BFFF00",
    "#39FF14", "#00FF9F", "#00F5D4", "#0AFFEF", "#00E5FF",
    "#7DF9FF", "#BF00FF", "#FF10F0", "#FE019A", "#FF6EC7", "#1B03A3",
  ]

  static var projectColors: [Color] {
    projectHex.map { AppColors.parseHex($0) }
  }

  static let defaultHex = "#63C7D8"
}

File: stacked-web/lib/theme/palette-colors.ts
/** Paleta unificada — paridade com Flutter `lib/theme/palette_colors.dart` */
export const PALETTE_HEX = [
  // Tons suaves
  "#63C7D8",
  "#6F8FB8",
  "#84B98E",
  "#789C6B",
  "#C58D97",
  "#C58A72",
  "#A496C8",
  "#6F79B6",
  "#C7B38A",
  "#D3B36A",
  "#7F99A8",
  "#9CA3AF",
  // Tons fortes (design system)
  "#5FD3DC",
  "#4D9FEC",
  "#B18CF5",
  "#8FD46B",
  "#F5A623",
  "#EF5A5F",
  "#FF85A1",
  "#64D8A0",
  "#FFD166",
  "#E07B54",
  // Vibrantes
  "#F43F5E",
  "#EC4899",
  "#D946EF",
  "#06B6D4",
  "#10B981",
  "#84CC16",
  "#F59E0B",
  "#F97316",
  // Variantes escuras
  "#0E7490",
  "#1D4ED8",
  "#6D28D9",
  "#15803D",
  "#B45309",
  "#B91C1C",
  "#BE185D",
  "#047857",
  "#0F766E",
  "#1E40AF",
  "#5B21B6",
  "#166534",
  "#92400E",
  "#7F1D1D",
  "#831843",
  "#134E4A",
] as const;

/** Primeiras 12 cores — preview compacto no seletor */
export const PALETTE_PREVIEW_HEX = PALETTE_HEX.slice(0, 12);

export const DEFAULT_PALETTE_HEX = PALETTE_HEX[0];
